### PR TITLE
JSDoc fixes

### PR DESCRIPTION
**Pull request by ntwb**
## Description

A bunch of JSDoc fixes

## How Has This Been Tested?

Running a _local_ ESLint configuration based on what was removed in #4245 

This resolves most `eslint-plugin-jsdoc` styleistic rules, it doesn't add new JSDocs.

A good way to review this PR is to use the "split view" and search for two spaces, this should highlight the whitespace changes well for the right hand column.

## Types of changes

JSDoc updates

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

File: blocks/media-upload/index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const getGalleryDetailsMediaFrame = () => {
 		/**
 		 * Create the default states.
 		 *
-		 * @returns {void}
+		 * @returns {void} Void.
```
Comment (atimmer) [inline blocks/media-upload/index.js]: This is not necessary. The `{void}` return type removes the need for a return description.

File: blocks/api/serializer.js [hunk 2]
```diff
@@ -98,8 +98,8 @@ export function getSaveContent( blockType, attributes ) {
  * This function returns only those attributes which are needed to persist and
  * which cannot be matched from the block content.
  *
- * @param   {Object<String,*>} allAttributes Attributes from in-memory block data
```
Comment (atimmer) [inline blocks/api/serializer.js]: These `String`'s should also be lowercase.

File: blocks/api/registration.js [hunk 3]
```diff
@@ -203,14 +203,14 @@ export function getBlockTypes() {
...
 
 /**
- * Returns true if the block defines support for a feature, or false otherwise
+ * Returns true if the block defines support for a feature, or false otherwise.
  *
- * @param  {(String|Object)} nameOrType      Block name or type object
- * @param  {String}          feature         Feature to test
- * @param  {Boolean}         defaultSupports Whether feature is supported by
- *                                           default if not explicitly defined
+ * @param {(String|Object)} nameOrType      Block name or type object.
```
Comment (atimmer) [inline blocks/api/registration.js]: This `String` should also be lowercased.

File: components/autocomplete/test/index.js [hunk 4]
```diff
@@ -77,9 +78,12 @@ function par( /* arguments */ ) {
...
  * an input event. It also updates the data-cursor attribute which is used to
  * simulate the cursor position in the test mocks.
- * @param {*} wrapper enzyme wrapper around react node containing a FakeEditor.
- * @param {Array.<Node>} nodeList array of dom nodes.
- * @param {Array.<Number>} cursorPosition array specifying the child indexes and offset of the cursor
+ *
+ * @param {*}              wrapper        Enzyme wrapper around react node
+ *                                        containing  a FakeEditor.
+ * @param {Array.<Node>}   nodeList       Array of dom nodes.
+ * @param {Array.<Number>} cursorPosition Array specifying the child indexes and
```
Comment (atimmer) [inline components/autocomplete/test/index.js]: This `Number` should also be lowercase.

File: data/index.js [hunk 5]
```diff
@@ -66,9 +66,10 @@ export function registerSelectors( reducerKey, newSelectors ) {
  * Higher Order Component used to inject data using the registered selectors.
  *
  * @param {Function} mapSelectorsToProps Gets called with the selectors object
- *                                       to determine the data for the component.
+ *                                       to determine the data for the
+ *                                       component.
  *
- * @returns {Func}                       Renders the wrapped component and passes it data.
+ * @returns {Func} Renders the wrapped component and passes it data.
```
Comment (atimmer) [inline data/index.js]: This should be `Function`.

File: editor/store/actions.js [hunk 6]
```diff
@@ -160,8 +161,8 @@ export function toggleSelection( isSelectionEnabled = true ) {
  * Returns an action object signalling that a blocks should be replaced with
  * one or more replacement blocks.
  *
- * @param  {(String|String[])} uids   Block UID(s) to replace
- * @param  {(Object|Object[])} blocks Replacement block(s)
+ * @param {(String|String[])} uids   Block UID(s) to replace.
```
Comment (atimmer) [inline editor/store/actions.js]: This `String` should be lowercase.

File: editor/store/actions.js [hunk 7]
```diff
@@ -177,8 +178,8 @@ export function replaceBlocks( uids, blocks ) {
  * Returns an action object signalling that a single block should be replaced
  * with one or more replacement blocks.
  *
- * @param  {(String|String[])} uid   Block UID(s) to replace
- * @param  {(Object|Object[])} block Replacement block(s)
+ * @param {(String|String[])} uid   Block UID(s) to replace.
```
Comment (atimmer) [inline editor/store/actions.js]: These `String`'s should be lowercase.

File: editor/store/actions.js [hunk 8]
```diff
@@ -345,10 +346,12 @@ export function stopTyping() {
...
 /**
- * Returns an action object used in signalling that the user toggled the sidebar
+ * Returns an action object used in signalling that the user toggled the
+ * sidebar.
  *
- * @param  {String}   sidebar      Name of the sidebar to toggle (desktop, mobile or publish)
- * @param  {Boolean?} forcedValue  Force a sidebar state
+ * @param {string}   sidebar     Name of the sidebar to toggle
+ *                               (desktop, mobile or publish).
+ * @param {Boolean?} forcedValue Force a sidebar state.
```
Comment (atimmer) [inline editor/store/actions.js]: This `Boolean` should be lowercase.

File: editor/utils/dom.js [hunk 9]
```diff
@@ -200,9 +200,9 @@ export function placeCaretAtHorizontalEdge( container, isReverse ) {
  *
  * @see https://developer.mozilla.org/en-US/docs/Web/API/Document/caretRangeFromPoint
  *
- * @param  {Document} doc The document of the range.
- * @param  {Float}    x   Horizontal position within the current viewport.
- * @param  {Float}    y   Vertical position within the current viewport.
+ * @param {Document} doc The document of the range.
+ * @param {Float}    x   Horizontal position within the current viewport.
```
Comment (atimmer) [inline editor/utils/dom.js]: `Float` is not a thing in JavaScript. This should be `number`.

File: editor/utils/dom.js [hunk 10]
```diff
@@ -229,10 +229,10 @@ function caretRangeFromPoint( doc, x, y ) {
  * Gives the container a temporary high z-index (above any UI).
  * This is preferred over getting the UI nodes and set styles there.
  *
- * @param  {Document} doc       The document of the range.
- * @param  {Float}    x         Horizontal position within the current viewport.
- * @param  {Float}    y         Vertical position within the current viewport.
- * @param  {Element}  container Container in which the range is expected to be found.
+ * @param {Document} doc       The document of the range.
+ * @param {Float}    x         Horizontal position within the current viewport.
```
Comment (atimmer) [inline editor/utils/dom.js]: `Float`.

**Review by atimmer — CHANGES_REQUESTED** [review]
This looks good, I have several minor suggestions.

**Comment by ntwb** [comment]
Thanks @atimmer, fixed in https://github.com/WordPress/gutenberg/pull/4563/commits/968e87f36023fc62f8163b7c4a92b2bf8ee09986 👍 

I'll create a few upstream issues for the above, `eslint-plugin-jsdoc` should have picked up most of these I would have thought.

**Review by atimmer — APPROVED** [review]

**Pull request merged**